### YamlMime:ManagedReference
items:
- uid: System.Data.Linq.DuplicateKeyException
  id: DuplicateKeyException
  children:
  - System.Data.Linq.DuplicateKeyException.#ctor(System.Object)
  - System.Data.Linq.DuplicateKeyException.#ctor(System.Object,System.String)
  - System.Data.Linq.DuplicateKeyException.#ctor(System.Object,System.String,System.Exception)
  - System.Data.Linq.DuplicateKeyException.Object
  langs:
  - csharp
  name: DuplicateKeyException
  nameWithType: DuplicateKeyException
  fullName: System.Data.Linq.DuplicateKeyException
  type: Class
  summary: "Thrown when an attempt is made to add an object to the identity cache by using a key that is already being used."
  syntax:
    content: 'public class DuplicateKeyException : InvalidOperationException'
  inheritance:
  - System.InvalidOperationException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Data.Linq.DuplicateKeyException.#ctor(System.Object)
  id: '#ctor(System.Object)'
  parent: System.Data.Linq.DuplicateKeyException
  langs:
  - csharp
  name: DuplicateKeyException(Object)
  nameWithType: DuplicateKeyException.DuplicateKeyException(Object)
  fullName: System.Data.Linq.DuplicateKeyException.DuplicateKeyException(Object)
  type: Constructor
  assemblies:
  - System.Data.Linq
  namespace: System.Data.Linq
  summary: "Initializes a new instance of the <xref href=&quot;System.Data.Linq.DuplicateKeyException&quot;></xref> class."
  syntax:
    content: public DuplicateKeyException (object duplicate);
    parameters:
    - id: duplicate
      type: System.Object
      description: "The duplicate key that caused the exception to be thrown."
  overload: System.Data.Linq.DuplicateKeyException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Data.Linq.DuplicateKeyException.#ctor(System.Object,System.String)
  id: '#ctor(System.Object,System.String)'
  parent: System.Data.Linq.DuplicateKeyException
  langs:
  - csharp
  name: DuplicateKeyException(Object,String)
  nameWithType: DuplicateKeyException.DuplicateKeyException(Object,String)
  fullName: System.Data.Linq.DuplicateKeyException.DuplicateKeyException(Object,String)
  type: Constructor
  assemblies:
  - System.Data.Linq
  namespace: System.Data.Linq
  summary: "Initializes a new instance of the <xref href=&quot;System.Data.Linq.DuplicateKeyException&quot;></xref> class by referencing the duplicate key and providing an error message."
  syntax:
    content: public DuplicateKeyException (object duplicate, string message);
    parameters:
    - id: duplicate
      type: System.Object
      description: "The duplicate key that caused the exception to be thrown."
    - id: message
      type: System.String
      description: "The message to appear when the exception is thrown."
  overload: System.Data.Linq.DuplicateKeyException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Data.Linq.DuplicateKeyException.#ctor(System.Object,System.String,System.Exception)
  id: '#ctor(System.Object,System.String,System.Exception)'
  parent: System.Data.Linq.DuplicateKeyException
  langs:
  - csharp
  name: DuplicateKeyException(Object,String,Exception)
  nameWithType: DuplicateKeyException.DuplicateKeyException(Object,String,Exception)
  fullName: System.Data.Linq.DuplicateKeyException.DuplicateKeyException(Object,String,Exception)
  type: Constructor
  assemblies:
  - System.Data.Linq
  namespace: System.Data.Linq
  summary: "Initializes a new instance of the <xref href=&quot;System.Data.Linq.DuplicateKeyException&quot;></xref> class by referencing the duplicate key, providing an error message, and specifying the exception that caused this exception to be thrown."
  syntax:
    content: public DuplicateKeyException (object duplicate, string message, Exception innerException);
    parameters:
    - id: duplicate
      type: System.Object
      description: "The duplicate key that caused the exception to be thrown."
    - id: message
      type: System.String
      description: "The message to appear when the exception is thrown."
    - id: innerException
      type: System.Exception
      description: "The previous exception that caused the <xref href=&quot;System.Data.Linq.DuplicateKeyException&quot;></xref> exception to be thrown."
  overload: System.Data.Linq.DuplicateKeyException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Data.Linq.DuplicateKeyException.Object
  id: Object
  parent: System.Data.Linq.DuplicateKeyException
  langs:
  - csharp
  name: Object
  nameWithType: DuplicateKeyException.Object
  fullName: System.Data.Linq.DuplicateKeyException.Object
  type: Property
  assemblies:
  - System.Data.Linq
  namespace: System.Data.Linq
  summary: "Gets the object that caused the exception."
  syntax:
    content: public object Object { get; }
    return:
      type: System.Object
      description: "The object that caused the exception."
  overload: System.Data.Linq.DuplicateKeyException.Object*
  exceptions: []
  platform:
  - net462
references:
- uid: System.InvalidOperationException
  isExternal: false
  name: System.InvalidOperationException
- uid: System.Data.Linq.DuplicateKeyException.#ctor(System.Object)
  parent: System.Data.Linq.DuplicateKeyException
  isExternal: false
  name: DuplicateKeyException(Object)
  nameWithType: DuplicateKeyException.DuplicateKeyException(Object)
  fullName: System.Data.Linq.DuplicateKeyException.DuplicateKeyException(Object)
- uid: System.Object
  parent: System
  isExternal: true
  name: Object
  nameWithType: Object
  fullName: System.Object
- uid: System.Data.Linq.DuplicateKeyException.#ctor(System.Object,System.String)
  parent: System.Data.Linq.DuplicateKeyException
  isExternal: false
  name: DuplicateKeyException(Object,String)
  nameWithType: DuplicateKeyException.DuplicateKeyException(Object,String)
  fullName: System.Data.Linq.DuplicateKeyException.DuplicateKeyException(Object,String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Data.Linq.DuplicateKeyException.#ctor(System.Object,System.String,System.Exception)
  parent: System.Data.Linq.DuplicateKeyException
  isExternal: false
  name: DuplicateKeyException(Object,String,Exception)
  nameWithType: DuplicateKeyException.DuplicateKeyException(Object,String,Exception)
  fullName: System.Data.Linq.DuplicateKeyException.DuplicateKeyException(Object,String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Data.Linq.DuplicateKeyException.Object
  parent: System.Data.Linq.DuplicateKeyException
  isExternal: false
  name: Object
  nameWithType: DuplicateKeyException.Object
  fullName: System.Data.Linq.DuplicateKeyException.Object
- uid: System.Data.Linq.DuplicateKeyException.#ctor*
  parent: System.Data.Linq.DuplicateKeyException
  isExternal: false
  name: DuplicateKeyException
  nameWithType: DuplicateKeyException.DuplicateKeyException
- uid: System.Data.Linq.DuplicateKeyException.Object*
  parent: System.Data.Linq.DuplicateKeyException
  isExternal: false
  name: Object
  nameWithType: DuplicateKeyException.Object
